src/cache/cache_las.c

```diff
@@ -1101,65 +1110,85 @@ __wt_las_sweep(WT_SESSION_IMPL *session)
+                cursor->prev(cursor);
+
+            WT_ERR(cursor->get_key(cursor, &las_btree_id, &las_key, &las_timestamp, &las_txnid));
+            WT_ERR(
+              cursor->get_value(cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
+
+            /* Remove each individual LAS record till the current record. */
+            for (i = 0; i < skip_update_modify_cnt + 1; i++) {
+                __wt_verbose(session, WT_VERB_LOOKASIDE_ACTIVITY,
+                  "Sweep removing lookaside entry with "
+                  "btree ID: %" PRIu32 " saved key size: %" WT_SIZET_FMT ", record type: %" PRIu8
+                  " transaction ID: %" PRIu64,
+                  las_btree_id, saved_key->size, upd_type, las_txnid);
+
+                cursor->remove(cursor);
+                cursor->next(cursor);
+                WT_ERR(
+                  cursor->get_key(cursor, &las_btree_id, &las_key, &las_timestamp, &las_txnid));
+                WT_ERR(cursor->get_value(
+                  cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
+                ++remove_cnt;
+            }
+            skip_update_modify_cnt = 0;
+        }
 
+        /* Don't remove the non obsolete entry. */
+        if (las_timestamp > S2C(session)->txn_global.oldest_timestamp) {
             /*
-             * There are several conditions that need to be met
-             * before we choose to remove a key block:
-             *  * The entries were written with skew newest.
-             *    Indicated by the first entry being a birthmark.
-             *  * The first entry is globally visible.
-             *  * The entry wasn't from a prepared transaction.
+             * TODO: In case if there exists any skipped update modify records that can be removed,
+             * it is better to squash them to frame a UPDATE record and remove the obsolete records.
              */
-            if (upd_type == WT_UPDATE_BIRTHMARK &&
-              __wt_txn_visible_all(session, las_txnid, durable_timestamp) &&
-              prepare_state != WT_PREPARE_INPROGRESS)
-                removing_key_block = true;
-            else
-                removing_key_block = false;
-        }
-
-        if (!removing_key_block)
+            prev_entry_can_remove = false;
+            skip_update_modify_cnt = 0;
             continue;
+        }
 
-        __wt_verbose(session, WT_VERB_LOOKASIDE_ACTIVITY,
-          "Sweep removing lookaside entry with "
-          "btree ID: %" PRIu32 " saved key size: %" WT_SIZET_FMT " record type: %" PRIu8
-          " transaction ID: %" PRIu64,
-          las_btree_id, saved_key->size, upd_type, las_txnid);
-        WT_ERR(cursor->remove(cursor));
-        ++remove_cnt;
+        prev_entry_can_remove = true;
+        /* Save the first key. */
+        if (saved_btree_id == 0) {
+            saved_btree_id = las_btree_id;
+            WT_ERR(__wt_buf_set(session, saved_key, las_key.data, las_key.size));
+        }
```

Comment:
There's some code around line 1201:
```
/*  
 * If the loop terminates after completing a work unit, we will continue the table sweep next
 * time. Get a local copy of the sweep key, we're going to reset the cursor; do so before
 * calling cursor.remove, cursor.remove can discard our hazard pointer and the page could be
 * evicted from underneath us. 
 */  
if (ret == 0) {
    WT_ERR(__wt_cursor_get_raw_key(cursor, sweep_key));
    if (!WT_DATA_IN_ITEM(sweep_key))
        WT_ERR(__wt_buf_set(session, sweep_key, sweep_key->data, sweep_key->size));
}
```
I'm not sure what `cursor.remove` that comment is talking about and looking through old versions of the code, it seems like there was one near here awhile ago, but it's long gone. If you agree with that analysis, I'd suggest something like:
```
 * If the loop terminates after completing a work unit, we will continue the table sweep next
 * time. Get a local copy of the sweep key to initialize the cursor position.
```

Reply:
Fixed.